diff --git a/bazel/grpc_extra_deps.bzl b/bazel/grpc_extra_deps.bzl
index e57aa968ea..0191906397 100644
--- a/bazel/grpc_extra_deps.bzl
+++ b/bazel/grpc_extra_deps.bzl
@@ -53,10 +53,6 @@ def grpc_extra_deps(ignore_version_differences = False):
 
     api_dependencies()
 
-    go_rules_dependencies()
-    go_register_toolchains(version = "1.18")
-    gazelle_dependencies()
-
     # Pull-in the go 3rd party dependencies for protoc_gen_validate, which is
     # needed for building C++ xDS protos
     go_third_party()