diff --git a/skylib/kustomize/kustomize.bzl b/skylib/kustomize/kustomize.bzl
index 644f629..a64682d 100644
--- a/skylib/kustomize/kustomize.bzl
+++ b/skylib/kustomize/kustomize.bzl
@@ -18,13 +18,15 @@ load("//skylib:stamp.bzl", "stamp")
 _binaries = {
     "darwin_amd64": ("https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize%2Fv4.5.3/kustomize_v4.5.3_darwin_amd64.tar.gz", "b0a6b0568273d466abd7cd535c556e44aa9ff5f54c07e86ed9f3016b416de992"),
     "linux_amd64": ("https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize%2Fv4.5.3/kustomize_v4.5.3_linux_amd64.tar.gz", "e4dc2f795235b03a2e6b12c3863c44abe81338c5c0054b29baf27dcc734ae693"),
+    "linux_arm64": ("https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize%2Fv4.5.3/kustomize_v4.5.3_linux_arm64.tar.gz", "97cf7d53214388b1ff2177a56404445f02d8afacb9421339c878c5ac2c8bc2c8"),
+    "darwin_arm64": ("https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize%2Fv4.5.3/kustomize_v4.5.3_darwin_arm64.tar.gz", "2fb58138c319d404e1604ae6665356e211b2ea45f17f174df1322de0100a55c4")
 }
 
 def _download_binary_impl(ctx):
     if ctx.os.name == "linux":
-        platform = "linux_amd64"
+        platform = "linux_arm64"
     elif ctx.os.name == "mac os x":
-        platform = "darwin_amd64"
+        platform = "darwin_arm64"
     else:
         fail("Platform " + ctx.os.name + " is not supported")
 
